src/main/java/io/jenkins/plugins/gitlabserver/servers/GitLabServer.java

```diff
@@ -139,6 +135,26 @@ public String getCredentialsId() {
         return credentialsId;
     }
 
+    /**
+     * Data Bound Constructor
+     *
+     * @param name          A unique name to use to describe the end-point, if empty replaced with a random name
+     * @param serverUrl     The URL of this GitLab Server
+     * @param manageHooks   {@code true} if and only if Jenkins is supposed to auto-manage hooks for this end-point.
+     * @param credentialsId The {@link StandardUsernamePasswordCredentials#getId()} of the credentials to use for
+     *                      auto-management of hooks.
+     */
+    @DataBoundConstructor
+    public GitLabServer(@Nonnull String name, @NonNull String serverUrl, boolean manageHooks,
```

Comment:
Be careful of using constructors with too many parameters.
You will have to maintain binary compatibility when you change the signature.
Also, databound constructor should only be used for "Mandatory" parameters.
https://jenkins.io/doc/developer/plugin-development/pipeline-integration/#constructor-vs-setters

I'd suggest having serverURL as the mandatory parameter.
And move the rest to databound setters.

Reply:
Copy that.